sympy/solvers/tests/test_recurr.py

```diff
@@ -219,3 +219,14 @@ def test_issue_18751():
 def test_issue_15751():
     f = y(n) + 21*y(n + 1) - 273*y(n + 2) - 1092*y(n + 3) + 1820*y(n + 4) + 1092*y(n + 5) - 273*y(n + 6) - 21*y(n + 7) + y(n + 8)
     assert rsolve(f, y(n)) is not None
+
+
+def test_issue_17990():
+    f = -10*y(n) + 4*y(n + 1) + 6*y(n + 2) + 46*y(n + 3)
+    sol = rsolve(f, y(n))
+
+    expected = C0*((86*18**(S(1)/3)/69 + (-12 + (-1 + sqrt(3)*I)*(290412 + 3036*sqrt(9165))**(S(1)/3))*(1 - sqrt(3)*I)*(24201 + 253*sqrt(9165))**(S(1)/3)/276)/((1 - sqrt(3)*I)*(24201 + 253*sqrt(9165))**(S(1)/3)))**n + C1*((86*18**(S(1)/3)/69 + (-12 + (-1 - sqrt(3)*I)*(290412 + 3036*sqrt(9165))**(S(1)/3))*(1 + sqrt(3)*I)*(24201 + 253*sqrt(9165))**(S(1)/3)/276)/((1 + sqrt(3)*I)*(24201 + 253*sqrt(9165))**(S(1)/3)))**n + C2*(-43*18**(S(1)/3)/(69*(24201 + 253*sqrt(9165))**(S(1)/3)) - S(1)/23 + (290412 + 3036*sqrt(9165))**(S(1)/3)/138)**n
+    assert sol == expected
+
+    e = sol.subs({C0:1, C1:1, C2: 1, n: 1}).evalf(15)
```

Comment:
The reason for using the same value of 1 for C0, C1, C2 is because I verified this with mathematica whose rsolve implementation returns a slightly different result that only matches when the constants are swapped (probably they're iterating over the roots in a different order). That is, `expected` equals `mma_expected = sympify("C1/(23/(-1 - 43*(2/(3*(24201 + 253*sqrt(9165))))^(1/3) + ((1/2)*(24201 + 253*sqrt(9165)))^(1/3)/3^(2/3)))^n + (-(1/23) - ((1 + I*sqrt(3))*((1/2)*(24201 + 253*sqrt(9165)))^(1/3))/(46*3^(2/3)) + (43*(1 - I*sqrt(3)))/(23*2^(2/3)*(3*(24201 + 253*sqrt(9165)))^(1/3)))^n*C2 + (-(1/23) - ((1 - I*sqrt(3))*((1/2)*(24201 + 253*sqrt(9165)))^(1/3))/(46*3^(2/3)) + (43*(1 + I*sqrt(3)))/(23*2^(2/3)*(3*(24201 + 253*sqrt(9165)))^(1/3)))^n*C3", locals={"C1":C2, "C2": C1, "C3": C0, "n": n})`. Keeping all constants the same ensures this test won't break if sympy's implementation changes slightly. To avoid testing the string value `evalf` needs `chop=True` (unless there's a better way). I'm also guessing that if anyone can come up with soluble integer initial conditions for the recurrence it might increase the testcase runtime significantly due to the `solve` call.